### PR TITLE
[CS/Refactor] restructure code to MVP pattern @open sesame 09/24 19:32 

### LINKED ISSUE #577: [CS] Restructure the code
As the application grows bigger, the application is becoming messier.

I feel managing the code is going slightly out of hand.

So. restructuring the application is needed.

Current thinking is using some kind of MVP pattern.

1. add util header to accommodate simple function, generate...
2. edje controls user interaction (simple interaction is handled here as well).
3. `view` creates view with registering callbacks(reside in `presenter`)
4. `presenter` delegates a logic to `view/edje` and `data`
5. `data` manipulates data.

The patch would just restructure the code so it will take less than a day.

### PR DESCRIPTION
**Pull request by zhoonit**
This patch layouts Customshortcut refactor. It is being migrated
Model-View-Presenter pattern for clarity.

**Changes proposed in this PR:**
- Add `presenter_*` to main presenter function
- Add `util_*` to data for utility function
- Some static function residing in `view.c` has exposed
- Strict segregation between view and data change

Resolves #577

**Self evaluation:**
1. Build test: [X]Passed [ ]Failed [ ]Skipped
2. Run test: [X]Passed [ ]Failed [ ]Skipped

Signed-off-by: Jihoon Lee <jhoon.it.lee@samsung.com>

### CODE REVIEW

**Comment by zhoonit** [comment]
## semantics:
** view handles view only **
creating, setting views are done. (controlling views are done in edje)
callback only related to view is registered inside `view/view`

** data handles data only **
data is changed at `data_*` only. (except very simple data like ad->tried`

** presenter manipulates data and view **
callback that invovles changing data is registered at `main/presenter`
presenter _has_ view and data

## reposition:
`data/data_parse_route` -> `data/util_parse_route`
`data/data_get_resource_path` -> `data/util_get_resource_path`
`view/process_train_result` -> `data/data_update_train_result`
`view/on_draw_proceed_infer_`  -> `main/presenter_on_canvas_submit_inference` 
`view/on_draw_proceed_training_`  -> `main/presenter_on_canvas_submit_training` 
`view/train` -> `presenter/train_` (facade)
`view/create_canvas_` -> `view/view_init_canvas`

## new methods:
`data_update_draw_target`
`data_handle_path_data` (page spsecific data initiation)
`presenter/init_page_` (page specific view initiataion, facade)

## rearranged :
`view/on_routes_to_` + `view/routes_to` -> `main/presenter_on_routes_request` + `presenter/routes_to_`(facade)
`view/set_draw_texts_` + `view/canvas_erase_all_` -> `view/view_set_canvas_clean`

**Review by jijoongmoon — APPROVED** [review]
LGTM

File: Applications/Tizen_native/CustomShortcut/inc/data.h [hunk 1]
```diff
@@ -111,27 +112,50 @@ int data_parse_route(const char *source, char **route, char **data);
...
+ * @brief handle given path_data. If data is invalid, it is essentially noop
+ *
+ * @param ad appdata
+ * @param data path_data to be handled
+ */
+void data_handle_path_data(appdata_s *ad, const char *data);
+
+/**
+ * @brief update draw target from the data. currently the label is depending on
+ * ad->tries and
```
Comment (kparichay) [inline Applications/Tizen_native/CustomShortcut/inc/data.h:127]: and?

Comment (zhoonit) [reply Applications/Tizen_native/CustomShortcut/inc/data.h]: Fixed.

**Review by kparichay — APPROVED** [review]
LGTM

**Pull request merged**